File: src/main/java/org/example/model/DbResult.java
package org.example.model;

import java.time.LocalDateTime;

public class DbResult {

    private LocalDateTime timestamp;
    private double dbConnectTimeMs;
    private double queryTimeMs;
    private boolean success;
    private String errorMessage;

    public DbResult() {
    }

    public DbResult(LocalDateTime timestamp,
                    double dbConnectTimeMs,
                    double queryTimeMs,
                    boolean success,
                    String errorMessage) {
        this.timestamp = timestamp;
        this.dbConnectTimeMs = dbConnectTimeMs;
        this.queryTimeMs = queryTimeMs;
        this.success = success;
        this.errorMessage = errorMessage;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public double getDbConnectTimeMs() {
        return dbConnectTimeMs;
    }

    public void setDbConnectTimeMs(double dbConnectTimeMs) {
        this.dbConnectTimeMs = dbConnectTimeMs;
    }

    public double getQueryTimeMs() {
        return queryTimeMs;
    }

    public void setQueryTimeMs(double queryTimeMs) {
        this.queryTimeMs = queryTimeMs;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    @Override
    public String toString() {

        StringBuilder builder = new StringBuilder();

        builder.append("DbResult{")
                .append("timestamp=").append(timestamp)
                .append(", dbConnectTimeMs=").append(dbConnectTimeMs)
                .append(", queryTimeMs=").append(queryTimeMs)
                .append(", success=").append(success);

        if (!success && errorMessage != null) {
            builder.append(", errorMessage='")
                    .append(errorMessage)
                    .append('\'');
        }

        builder.append('}');

        return builder.toString();
    }
}

File: node-version/src/model/DbResult.js
class DbResult {

    constructor(
        timestamp,
        dbConnectTimeMs,
        queryTimeMs,
        success,
        errorMessage = null
    ) {
        this.timestamp = timestamp;
        this.dbConnectTimeMs = dbConnectTimeMs;
        this.queryTimeMs = queryTimeMs;
        this.success = success;
        this.errorMessage = errorMessage;
    }

    toString() {

        const result = {
            timestamp: this.timestamp,
            dbConnectTimeMs: this.dbConnectTimeMs,
            queryTimeMs: this.queryTimeMs,
            success: this.success
        };

        if (!this.success && this.errorMessage) {
            result.errorMessage = this.errorMessage;
        }

        return JSON.stringify(result);
    }
}

export default DbResult;